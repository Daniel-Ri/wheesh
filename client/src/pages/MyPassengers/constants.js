export const SET_MY_PASSENGERS = 'MyPassengers/SET_MY_PASSENGERS';
export const GET_MY_PASSENGERS = 'MyPassengers/GET_MY_PASSENGERS';
