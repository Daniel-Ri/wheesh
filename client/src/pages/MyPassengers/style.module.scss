@import '@styles/base';

// Reset
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: Poppins;
}

.main {
  margin-left: 200px;
  position: relative;
  background: $color-alabaster;
  display: flex;
  justify-content: center;
  width: calc(100vw - 200px);
  height: 100%;
  min-height: 100vh;

  .container {
    width: 90%;
    max-width: 40rem;
    @include directionY;

    header {
      width: 100%;
      margin-top: 5vh;
      @include directionX;
      justify-content: center;
      align-items: center;
      position: relative;

      h1 {
        font-size: 120%;
      }
    }

    .message {
      margin-top: 1rem;
      width: 100%;
      border-radius: 10px;
      padding: 0.5rem;
      font-size: 90%;
      background-image: linear-gradient(to right, $color-picton-blue-light-shade, $color-white);
    }

    .addBtn {
      margin-top: 1rem;
      @include directionX;
      align-items: center;
      gap: 0.5rem;
      border-radius: 10px;
    }

    .passengers {
      margin-top: 1rem;
      margin-bottom: 1rem;
      width: 100%;
      @include directionY;
      gap: 1rem;
    }
  }

  @media screen and (max-width: $screen-lg-limit) {
    margin-left: 192px;
    width: calc(100vw - 192px);
  }

  @media screen and (max-width: $screen-md-limit) {
    margin-left: 10rem;
    width: calc(100vw - 10rem);
  }

  @media screen and (max-width: $screen-sm-limit) {
    margin-left: 0;
    width: 100vw;
    margin-bottom: 76px;
    height: 100%;
    min-height: calc(100vh - 76px);

    .container {
      header {
        height: 3rem;
      }
    }
  }
}