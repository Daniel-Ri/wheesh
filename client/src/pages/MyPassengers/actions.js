import { GET_MY_PASSENGERS, SET_MY_PASSENGERS } from './constants';

export const setMyPassengers = (myPassengers) => ({
  type: SET_MY_PASSENGERS,
  myPassengers,
});

export const getMyPassengers = () => ({
  type: GET_MY_PASSENGERS,
});
