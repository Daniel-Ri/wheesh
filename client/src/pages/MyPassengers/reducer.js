import { produce } from 'immer';
import { SET_MY_PASSENGERS } from './constants';

export const initialState = {
  myPassengers: [],
};

export const storedKey = [];

const myPassengersReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_MY_PASSENGERS:
        draft.myPassengers = action.myPassengers;
        break;
    }
  });

export default myPassengersReducer;
