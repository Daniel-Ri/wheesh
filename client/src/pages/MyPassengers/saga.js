import { takeLatest, call, put } from 'redux-saga/effects';
import { getMyPassengers } from '@domain/api';
import { GET_MY_PASSENGERS } from './constants';
import { setMyPassengers } from './actions';

function* doGetMyPassengers() {
  try {
    const response = yield call(getMyPassengers);
    yield put(setMyPassengers(response.data));
  } catch (error) {
    // error
  }
}

export default function* myPassengersSaga() {
  yield takeLatest(GET_MY_PASSENGERS, doGetMyPassengers);
}
