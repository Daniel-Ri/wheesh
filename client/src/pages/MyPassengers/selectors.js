import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectMyPassengersState = (state) => state.myPassengers || initialState;

export const selectMyPassengers = createSelector(selectMyPassengersState, (state) => state.myPassengers);
