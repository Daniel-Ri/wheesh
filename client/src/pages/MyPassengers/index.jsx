import { connect, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import BackBtn from '@components/BackBtn';
import { useEffect } from 'react';
import { createStructuredSelector } from 'reselect';
import { Button } from '@mui/material';
import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
import { useNavigate } from 'react-router-dom';
import { getMyPassengers } from './actions';
import { selectMyPassengers } from './selectors';
import PassengerCard from './components/PassengerCard';

import classes from './style.module.scss';

const MyPassengers = ({ myPassengers }) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(getMyPassengers());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <main data-testid="MyPassengers" className={classes.main}>
      <div className={classes.container}>
        <header>
          <BackBtn handleClickBack={() => navigate('/me')} />
          <h1>
            <FormattedMessage id="app_my_passengers" />
          </h1>
        </header>
        <div className={classes.message}>
          {myPassengers.length}/15 <FormattedMessage id="app_up_to_15_passengers" />
        </div>
        {myPassengers.length < 15 && (
          <Button variant="outlined" className={classes.addBtn} onClick={() => navigate('/createPassenger')}>
            <AddCircleOutlineIcon />
            <div className={classes.name}>
              <FormattedMessage id="app_add_passenger" />
            </div>
          </Button>
        )}
        <div className={classes.passengers}>
          {myPassengers.map((passenger) => (
            <PassengerCard key={passenger.id} passenger={passenger} />
          ))}
        </div>
      </div>
    </main>
  );
};

MyPassengers.propTypes = {
  myPassengers: PropTypes.array,
};

const mapStateToProps = createStructuredSelector({
  myPassengers: selectMyPassengers,
});

export default connect(mapStateToProps)(MyPassengers);
