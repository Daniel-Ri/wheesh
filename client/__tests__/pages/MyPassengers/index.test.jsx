import { render } from '@utils/testHelper';
import MyPassengers from '@pages/MyPassengers';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let wrapper;

const mockProps = {
  myPassengers: [
    {
      id: 1,
      userId: 1,
      isUser: true,
      gender: 'Male',
      dateOfBirth: '1995-11-11T17:00:00.000Z',
      idCard: '1234567890123456',
      name: 'John Doe',
      email: 'johndoe@gmail.com',
    },
    {
      id: 2,
      userId: 1,
      isUser: false,
      gender: 'Female',
      dateOfBirth: '1997-01-31T17:00:00.000Z',
      idCard: '1234567890123496',
      name: 'Jean Doe',
      email: 'jeandoe@gmail.com',
    },
  ],
};

beforeEach(() => {
  wrapper = render(<MyPassengers {...mockProps} />);
});

describe('Page MyPassengers', () => {
  test('Renders MyPassengers page', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('MyPassengers')).toBeInTheDocument();
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
