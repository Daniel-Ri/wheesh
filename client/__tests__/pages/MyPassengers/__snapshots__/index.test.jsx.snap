// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page MyPassengers Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
        data-testid="MyPassengers"
      >
        <div
          class="container"
        >
          <header>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              data-testid="BackBtn"
              tabindex="0"
              type="button"
            >
              Kembali
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <h1>
              Penumpang Saya
            </h1>
          </header>
          <div
            class="message"
          >
            0
            /15 
            (Dapat ditambahkan hingga 15 penumpang)
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium addBtn css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddCircleOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            <div
              class="name"
            >
              Tambah Penumpang
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="passengers"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
      data-testid="MyPassengers"
    >
      <div
        class="container"
      >
        <header>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="BackBtn"
            tabindex="0"
            type="button"
          >
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h1>
            Penumpang Saya
          </h1>
        </header>
        <div
          class="message"
        >
          0
          /15 
          (Dapat ditambahkan hingga 15 penumpang)
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium addBtn css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          <div
            class="name"
          >
            Tambah Penumpang
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="passengers"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
